### PR TITLE
Restore `tf.reduce_sum` binding and add isolated tests

### PR DESCRIPTION
**Pull request by khatchad**
## Summary

The `tf.reduce_sum` / `tf.math.reduce_sum` putfield bindings on the `tensorflow` module's `import` method were lost during branch-267 work and never restored. Without them, `tf.reduce_sum(...)` call sites never resolve to `Ltensorflow/math/reduce_sum`, so `TensorGeneratorFactory.create` never instantiates `ReduceSum`, and the analysis silently treats the return as untyped.

There were no isolated tests for `tf.reduce_sum` in the repo, so no CI signal flagged the regression. The two existing usages in test data (`train.py:314`, `tensorboard_example.py:121`) are buried inside larger expressions in functions that aren't the subject of any test's tensor-count assertion, so the missing binding has been silently latent.

## Lifecycle Of The Regression

| Date | Commit | What |
|---|---|---|
| 2026-02-07 | `d9d6209e` "Fix tf.reshape shape inference and enable Reshape generator" | Added the `tf.reduce_sum` bindings + class as part of broader generator-coverage work |
| 2026-02-11 | `8e526936` "Put back more changes." | Removed both bindings and class |
| 2026-02-11 (same day) | `793ea260` "Test updates." (body: "Re-add some changes for `reduce_mean`, etc.") | Re-added the class only — bindings missed during a re-add focused on `reduce_mean` |
| 2026-04-25 | `5467c89a` (PR wala/ML#156) | Squash-merged branch-267 to master, carrying the class-without-binding state |

So the missing binding looks like an oversight from a rapid same-day cleanup focused on `reduce_mean`, not a deliberate architectural choice.

## Test Plan

- [x] On `master`, the new `testReduceSum*` methods fail with `expected:<1> but was:<0>` (the analysis sees zero tensor parameters in `f`/`g`/`h` because the `reduce_sum` return isn't tracked as a tensor).
- [x] With this PR, all three pass.

Reproduce:

```bash
mvn -pl com.ibm.wala.cast.python.ml.test -am test \
    -Dtest='TestTensorflow2Model#testReduceSum*' \
    -Dsurefire.failIfNoSpecifiedTests=false
```

## Changes

- **`com.ibm.wala.cast.python.ml/data/tensorflow.xml`**: restore the three putfield lines on the `tensorflow` import method (alias under both `tensorflow` and `tensorflow.math`, mirroring the `reduce_mean` pattern). Added a comment explaining the alias rationale and the connection to `ReduceSum.java`.
- **`com.ibm.wala.cast.python.test/data/tf2_test_reduce_sum.py`**: new test data file mirroring `tf2_test_reduce_mean.py`'s three-case structure (scalar reduce, axis=0, axis=1).
- **`com.ibm.wala.cast.python.ml.test/.../TestTensorflow2Model.java`**: add `testReduceSum`, `testReduceSum2`, `testReduceSum3` modeled on `testReduceMean*`.

## Out Of Scope

- A broader audit of *other* TF API bindings that may have been similarly lost during branch-267 cleanup. There are likely more (the `read_data` marker pattern in classes that exist without bindings is a fingerprint to grep for). Tracked separately at [wala/ML#422](https://github.com/wala/ML/issues/422).

### CODE REVIEW

**Comment by khatchad** [comment]
@claude please review this PR.

File: com.ibm.wala.cast.python.ml.test/source/com/ibm/wala/cast/python/ml/test/TestTensorflow2Model.java [hunk 1]
```diff
@@ -3869,6 +3869,24 @@ public void testReduceMean8()
...
+    test("tf2_test_reduce_sum.py", "g", 1, 1, Map.of(2, Set.of(TENSOR_2_FLOAT32)));
+  }
+
+  @Test
+  public void testReduceSum3()
+      throws ClassHierarchyException, IllegalArgumentException, CancelException, IOException {
+    test("tf2_test_reduce_sum.py", "h", 1, 1, Map.of(2, Set.of(TENSOR_2_FLOAT32)));
+  }
+
   @Test
```
[the remark this answers is not in the document]
Comment (khatchad) [reply com.ibm.wala.cast.python.ml.test/source/com/ibm/wala/cast/python/ml/test/TestTensorflow2Model.java:3992]: Good catch — addressed in fd1c6201. Added `tf2_test_reduce_sum_kwargs.py` mirroring `tf2_test_reduce_mean_kwargs.py`'s structure (5 cases: kwargs, mixed pos/kw, `keepdims`, `tf.math.reduce_sum` with kwargs, `tf.constant` with `shape` kwarg) and `testReduceSum4`–`testReduceSum8` test methods. `testReduceSum7` specifically exercises `tf.math.reduce_sum(input_tensor=x, axis=1, keepdims=True)` so future regressions in the `ref="math"` binding fail there. All 8 `testReduceSum*` tests pass locally.
